CenterForOpenScience/osf.io
# [Feature] Add explicit year string elastic mapping [OSF-6992]

# This requires search migration

## Purpose

No license needs to allow year ranges. Elastic search auto infers the datatype of when no explicit mapping exists. Because years were mostly numbers, it assumed it was always numbers. Year ranges and other strings broke indexing. This explicitly maps license years to strings in elastic

## Changes

Explicitly map license years to strings.

## Side effects

None known.

## Ticket
[#OSF-6992]
https://openscience.atlassian.net/browse/OSF-6992


## QA Notes 
1. Note that the only fix going in is one that stops elastic from failing. Product was fine with no validation on the year field.
2. The main thing to check is that search still works properly. Minimal checks are fine since it is unlikely to break much.
3. Also test putting non-numeric values in the license picker and notify me so I can check sentry logs.
